thirdparty/faiss/faiss/IndexIVFFlat.cpp

```diff
@@ -514,6 +596,192 @@ void IndexIVFFlat::reconstruct_from_offset(
     memcpy(recons, invlists->get_single_code(list_no, offset), code_size);
 }
 
+std::unique_ptr<IVFFlatIteratorWorkspace> IndexIVFFlat::getIteratorWorkspace(
+        const float* query_data,
+        const IVFSearchParameters* ivfsearchParams) const {
+    return std::make_unique<IVFFlatIteratorWorkspace>(
+            query_data, ivfsearchParams);
+}
+
+std::optional<std::pair<float, idx_t>> IndexIVFFlat::getIteratorNext(
+        IVFFlatIteratorWorkspace* workspace) const {
+    bool interrupt = false;
+    std::mutex exception_mutex;
```

Comment:
mutex not necessary, iterator class is not designed thread safe. User must protect concurrent `iterator->Next()` calls themselves.